bundles/org.eclipse.equinox.p2.director.app/src/org/eclipse/equinox/internal/p2/director/app/DirectorApplication.java

```diff
@@ -1058,6 +1063,13 @@ private void performProvisioningActions() throws CoreException {
 			context.setProperty(ProvisioningContext.FOLLOW_REPOSITORY_REFERENCES, String.valueOf(followReferences));
 			context.setProperty(FOLLOW_ARTIFACT_REPOSITORY_REFERENCES, String.valueOf(followReferences));
 
+			if (addJREIU) {
+				List<IInstallableUnit> extraUnits = new ArrayList<>();
+				IInstallableUnit jreIU = JREAction.createJREIU();
+				extraUnits.add(jreIU);
+				context.setExtraInstallableUnits(extraUnits);
```

Comment:
I'm really splitting hairs now, but you could use List.of to create the list to simplify this to a single line of code.

Reply:
Fixed.